features/FEATURE_BLE/source/gap/AdvertisingDataBuilder.cpp

```diff
@@ -0,0 +1,469 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ble/gap/AdvertisingDataBuilder.h"
+
+namespace ble {
+
+AdvertisingDataBuilder::AdvertisingDataBuilder(mbed::Span<uint8_t> buffer) :
+    _buffer(buffer),
+    _payload_length(0) {
+}
+
+AdvertisingDataBuilder::AdvertisingDataBuilder(uint8_t* buffer, size_t buffer_size) :
+    _buffer(buffer, buffer_size),
+    _payload_length(0) {
+}
+
+mbed::Span<const uint8_t> AdvertisingDataBuilder::getAdvertisingData() const {
+    return _buffer.first(_payload_length);
+}
+
+ble_error_t AdvertisingDataBuilder::addData(
+    adv_data_type_t advDataType,
+    mbed::Span<const uint8_t> fieldData
+) {
+    if (findField(advDataType)) {
+        return BLE_ERROR_OPERATION_NOT_PERMITTED;
+    } else {
+        return addField(advDataType, fieldData);
+    }
+}
+
+ble_error_t AdvertisingDataBuilder::replaceData(
+    adv_data_type_t advDataType,
+    mbed::Span<const uint8_t> fieldData
+) {
+    uint8_t* field = findField(advDataType);
+
+    if (field) {
+        return replaceField(advDataType, fieldData, field);
+    } else {
+        return BLE_ERROR_NOT_FOUND;
+    }
+}
+
+ble_error_t AdvertisingDataBuilder::appendData(
+    adv_data_type_t advDataType,
+    mbed::Span<const uint8_t> fieldData
+) {
+    uint8_t* field = findField(advDataType);
+
+    if (field) {
+        return appendToField(fieldData, field);
+    } else {
+        return BLE_ERROR_NOT_FOUND;
+    }
+}
+
+ble_error_t AdvertisingDataBuilder::removeData(
+    adv_data_type_t advDataType
+) {
+    uint8_t* field = findField(advDataType);
+
+    if (field) {
+        return removeField(field);
+    } else {
+        return BLE_ERROR_NOT_FOUND;
+    }
+}
+
+ble_error_t AdvertisingDataBuilder::addOrReplaceData(
+    adv_data_type_t advDataType,
+    mbed::Span<const uint8_t> fieldData
+) {
+    uint8_t* field = findField(advDataType);
+
+    if (field) {
+        return replaceField(advDataType, fieldData, field);
+    } else {
+        return addField(advDataType, fieldData);
+    }
+}
+
+ble_error_t AdvertisingDataBuilder::addOrAppendData(
+    adv_data_type_t advDataType,
+    mbed::Span<const uint8_t> fieldData
+) {
+    uint8_t* field = findField(advDataType);
+
+    if (field) {
+        return appendToField(fieldData, field);
+    } else {
+        return addField(advDataType, fieldData);
+    }
+}
+
+void AdvertisingDataBuilder::clear() {
+    memset(_buffer.data(), 0, _buffer.size());
+    _payload_length = 0;
+}
+
+ble_error_t AdvertisingDataBuilder::setAppearance(
+    adv_data_appearance_t appearance
+) {
+    uint8_t appearence_byte = appearance.value();
+    mbed::Span<const uint8_t> appearance_span((const uint8_t*) &appearence_byte, 2);
```

Comment:
What does the 2 mean ? Probably should be commented or #defined 
Similarly with values below.

Reply:
It was incorrect to begin with. Fixed.